src/bun.js/node/node_fs.zig

```diff
@@ -3805,7 +3805,11 @@ pub const NodeFS = struct {
                     else => {
                         return .{ .err = err.withPath(this.osPathIntoSyncErrorBuf(path[0..len])) };
                     },
-                    .EXIST => return switch (bun.sys.directoryExistsAt(bun.invalid_fd, path)) {
+                    // mkpath_np in macOS also checks for EISDIR.
+                    .ISDIR,
```

Comment:
can we skip the `directoryExistsAt` if error is `EISDIR`?

Reply:
no, we have no idea why Apple added EISDIR in their mkpath_np since all they did was link to a radar URL, and so cargo culting is a safer choice